src/state/slices/foxEthSlice/selectors.ts

```diff
@@ -81,34 +83,35 @@ export const selectFoxEthLpOpportunityByAccountAddress = createSelector(
 export const selectFoxEthLpAccountsOpportunitiesAggregated = createDeepEqualOutputSelector(
   selectFoxEthLpAccountOpportunitiesByMaybeAccountAddress,
   (state: ReduxState) => state,
-  (wrappedEthLpOpportunities, state) => {
-    const aggregatedOpportunity = wrappedEthLpOpportunities
-      .filter(Boolean)
-      .reduce((acc, currentOpportunity) => {
-        acc = {
-          ...currentOpportunity,
-          underlyingFoxAmount: bnOrZero(acc.underlyingFoxAmount)
-            .plus(currentOpportunity.underlyingFoxAmount ?? '')
-            .toString(),
-          underlyingEthAmount: bnOrZero(acc.underlyingEthAmount)
-            .plus(currentOpportunity.underlyingEthAmount ?? '')
-            .toString(),
-          cryptoAmount: bnOrZero(acc.cryptoAmount)
-            .plus(currentOpportunity.cryptoAmount ?? '')
-            .toString(),
-          fiatAmount: bnOrZero(acc.fiatAmount)
-            .plus(currentOpportunity.fiatAmount ?? '')
-            .toString(),
-        }
-        return acc
-      }, {} as FoxEthLpEarnOpportunityType)
+  (wrappedEthLpOpportunities, state): FoxEthLpEarnOpportunityType => {
+    const aggregatedOpportunity = wrappedEthLpOpportunities.reduce<
+      Partial<FoxEthLpEarnOpportunityType | undefined>
+    >(
+      (acc, currentOpportunity) => ({
```

Comment:
i know that opportunities is a fairly short list, but just be careful with memory allocations using this reduce spread pattern

i fired a big footgun doing this with 6k+ assets in the asset service, was like 250ms blocking

Reply:
Check this out as an alternative pattern: https://github.com/shapeshift/web/pull/3255